lua/lean/lean3.lua

```diff
@@ -1,22 +1,52 @@
+local components = require('lean.infoview.components')
 local lean3 = {}
 
+-- Split a Lean 3 server response on goals.
+--
+-- Looks for ⊢, but ignores indented following lines for multi-line
```

Comment:
That's the correct heuristic.  (⊢ at the beginning of a line until an empty line would work as well)